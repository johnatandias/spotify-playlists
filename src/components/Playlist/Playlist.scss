@import '../../variables';

.playlistItem {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-right: 18px;
  margin-bottom: 18px;
  max-width: 150px;

  &__image {
    height: 150px;
    width: 150px;
  }

  &__name, &__amount {
    color: $primary-text-color;
  }

  &__name {
    font-size: 15px;
    font-weight: 500;
    margin-top: 10px;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
    max-width: 150px;
  }

  &__amount {
    font-size: 10px;
    font-weight: 400;
    margin-top: 4px;
  }
}
