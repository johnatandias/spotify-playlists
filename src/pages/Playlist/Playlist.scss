@import '../../variables';

.playlist {
  display: flex;
  height: 100%;
  width: 100%;
  background: $background-gradient;
  flex-direction: row;

  &__info {
    min-width: 280px;
    width: 280px;
    height: 100%;
    background-color: $background-color-playlist-info;
    display: flex;
    flex-direction: column;

    &__logo {
      margin-top: 50px;
      margin-left: 60px;
    }

    &__playlist {
      margin-top: 40px;
      margin-right: 0;
      align-self: center;
    }
  }

  &__content {
    display: flex;
    margin: 60px 80px 0;
    flex-direction: column;

    &__backNavigation {
      margin-bottom: 40px;
      min-height: 35px;
    }

    &__tracks {
      display: flex;
      padding-right: 15px;
      overflow-x: hidden;
      overflow-y: auto;
      white-space: nowrap;
      max-height: 100%;
    }
  }

  &__profile {
    align-self: flex-start;
    margin-top: 50px;
  }
}
