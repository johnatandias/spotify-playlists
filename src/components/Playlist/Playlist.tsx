import React from 'react';
import './Playlist.scss';

interface Playlist {
  name: string,
  amount: number,
  image: string,
  id?: string,
  onClick?: Function,
  className?: string
}

export const Playlist = ({ id, name, amount, image, onClick, className = '' }: Playlist) => (
  <div className={`playlistItem ${className}`}>
    <img
      className="playlistItem__image"
      onClick={() => onClick && onClick({ id, name, amount, image })}
      src={image}
      alt={name}
    />

    <span className="playlistItem__name" onClick={() => onClick && onClick({ id, name, amount, image })}>
      {name}
    </span>

    <span className="playlistItem__amount">
      {amount} {amount > 1 ? 'musics' : 'music'}
    </span>
  </div>
);
