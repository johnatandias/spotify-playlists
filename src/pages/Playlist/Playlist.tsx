import React, { useState, useEffect } from 'react';
import { RouteComponentProps } from 'react-router';
import { Logo } from '../../components/Logo/Logo';
import { Playlist as PlaylistInformation } from '../../components/Playlist/Playlist';
import { SearchBar } from '../../components/SearchBar/SearchBar';
import { Profile } from '../../components/Profile/Profile';
import { Back } from '../../components/Back/Back';
import { PlaylistItems, PlaylistItem } from '../../components/PlaylistItems/PlaylistItems';
import { SpotifyService } from '../../services/spotify';
import './Playlist.scss';

export const Playlist = ({ history }: RouteComponentProps) => {
  useEffect(() => {
    getPlaylistData();
    // eslint-disable-next-line
  }, []);

  const getPlaylistData = async () => {
    const musics = await SpotifyService.getPlaylistInformation(history.location.state.id);
    setMusics(musics);
  }

  const [musics, setMusics] = useState<PlaylistItem[]>([]);
  const [filtered, setFiltered] = useState<PlaylistItem[]>([]);

  const onSearch = (value: string) => {
    if (!value.length) return setFiltered([]);

    const items = musics.filter(({ name, artist }) => (
      name.toLowerCase().includes(value.toLowerCase()) ||
      artist.toLowerCase().includes(value.toLowerCase())
    ));

    setFiltered(items);
  }

  return (
    <div className="playlist">
      <div className="playlist__info">
        <Logo height={50} width={50} className="playlist__info__logo" />

        <PlaylistInformation
          name={history.location.state.name}
          image={history.location.state.image}
          amount={history.location.state.amount}
          className="playlist__info__playlist"
        />
      </div>

      <section className="playlist__content">
        <Back
          height={35}
          width={35}
          className="playlist__content__backNavigation"
          onClick={() => history.goBack()}
        />

        <SearchBar onSearch={onSearch} placeholder="Pesquisar..." />

        <PlaylistItems
          items={filtered.length ? filtered : musics}
          className="playlist__content__tracks"
        />
      </section>

      <Profile
        {...SpotifyService.getUserInformation()}
        className="playlist__profile"
        minimal
      />
    </div>
  );
};
