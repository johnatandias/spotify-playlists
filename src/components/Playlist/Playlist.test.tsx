import React from "react";
import { shallow } from 'enzyme';
import { Playlist } from './Playlist';

describe('Playlist component', () => {
  it('Renders without crashing', () => {
    const props = { id: 'id', name: 'name', amount: 10, image: 'image' };
    const wrapper = shallow(<Playlist {...props} />);
    expect(wrapper.find('img').prop('src')).toEqual(props.image);
    expect(wrapper.find('img').prop('alt')).toEqual(props.name);
    expect(wrapper.childAt(1).text().includes(props.name)).toBe(true);
    expect(wrapper.childAt(2).text().includes('10 musics')).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });

  it('Renders without crashing', () => {
    const props = { id: 'id', name: 'name', amount: 10, image: 'image', className: 'my-class' };
    const wrapper = shallow(<Playlist {...props} />);
    expect(wrapper.hasClass(props.className)).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should handle the click on image', () => {
    window.alert = jest.fn();
    const callAlert = (props: any) => alert(JSON.stringify(props));
    const props = { id: 'id', name: 'name', amount: 10, image: 'image' };

    const wrapper = shallow(<Playlist {...props} onClick={callAlert} />);
    wrapper.find('img').simulate('click');
    expect(window.alert).toHaveBeenCalledWith(JSON.stringify(props));
  });

  it('Should handle the click on playlist name', () => {
    window.alert = jest.fn();
    const callAlert = (props: any) => alert(JSON.stringify(props));
    const props = { id: 'id', name: 'name', amount: 10, image: 'image' };

    const wrapper = shallow(<Playlist {...props} onClick={callAlert} />);
    wrapper.childAt(1).simulate('click');
    expect(window.alert).toHaveBeenCalledWith(JSON.stringify(props));
  });
});
