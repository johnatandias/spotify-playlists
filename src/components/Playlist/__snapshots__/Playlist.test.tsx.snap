// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Playlist component Renders without crashing 1`] = `ShallowWrapper {}`;

exports[`Playlist component Renders without crashing 2`] = `ShallowWrapper {}`;
